src/Microsoft.NetCore.Analyzers/Core/Security/SourceTriggeredTaintedDataAnalyzerBase.cs

```diff
@@ -84,6 +84,18 @@ public override void Initialize(AnalysisContext context)
                                 },
                                 OperationKind.Invocation);
 
+                            operationBlockStartContext.RegisterOperationAction(
+                                operationAnalysisContext =>
```

Comment:
We could only register this operation action if the sourceInfoSymbolMap contains a SourceInfo with TaintArray.

Reply:
(And similarly only register operation actions if applicable.)

---
In reply to: [296963270](https://github.com/dotnet/roslyn-analyzers/pull/2624#discussion_r296963270) [](ancestors = 296963270)

Reply:
All the ConcreteInfos and InterfaceInfos in TaintedDataSymbolMap<TInfo> are private, I don't know how to find the sourceInfo with TaintArray in this case.

Reply:
You can change the ConcreteInfos and InterfaceInfos to internal, so you have access to the info you need.  🙂